packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -149,21 +153,27 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
 
   Future<void> _stopImageStream() async {
     stoppingCamera = true;
-    _controller?.pausePreview();
-    if (mounted) {
-      setState(() {});
-    }
+    setState(() {});
     await _controller?.dispose();
     _controller = null;
   }
 
   // Convert the [CameraImage] to a [InputImage] and checking this for barcodes
   // with help from ML Kit
   Future<void> _processCameraImage(CameraImage image) async {
+    //Only scanning every 10th image, but not resetting until the current one
+    //is done, so that we don't have idle time when the scanning takes longer
+    //than the 10 frames (approx. 1/3 second when 30fps)
+    if (frameCounter < 10) {
+      frameCounter++;
+      return;
+    }
```

Comment:
1. I would start immediately: here the very first scan is after 1/3 seconds.
2. I would use a `static const` with an explicit name, like `static const int _SKIPPED_FRAMES = 10;`

Reply:
How about 
```
if (framCounter % 10 != 0) {
  return;
}
frameCounter++;
```

Reply:
Agreed with @monsieurtanuki. Otherwise LG

Reply:
According to https://stackoverflow.com/a/50429767/14517119
> In Dart 2 int is limited to 64 bit

I think we're safe before we reach MAX_INT then ;)

@jasmeet0817 Your modulo suggestion does not work because once we decide to work we don't know how much time it will take - if it takes more than 1/3 seconds it's not good.

@M123-dev I see that you also use a `bool`, `isBusy`. I suggest that you use a 3-state `bool?` instead:
* `isBusy == false` means that you're ready to compute immediately
* `isBusy == true` means that you're working
* `isBusy == null` means that you're done with working and that you'll wait 10 frames before working again

Reply:
> According to https://stackoverflow.com/a/50429767/14517119
> 
> > In Dart 2 int is limited to 64 bit
> 
> I think we're safe before we reach MAX_INT then ;)
> 
> @jasmeet0817 Your modulo suggestion does not work because once we decide to work we don't know how much time it will take - if it takes more than 1/3 seconds it's not good.
> 
> @M123-dev I see that you also use a `bool`, `isBusy`. I suggest that you use a 3-state `bool?` instead:
> 
> * `isBusy == false` means that you're ready to compute immediately
> * `isBusy == true` means that you're working
> * `isBusy == null` means that you're done with working and that you'll wait 10 frames before working again

An enum would be better if we want more than 2 states:)

General questions:
1. If the processing takes more than 1/3rd sec, do we want to process next frame directly or wait 9 frames and then process.

Reply:
I'm against waiting until the latest I'd done and then counting, this could lead to very long waiting times

Reply:
@M123-dev Both @jasmeet0817 and I agreed on most of your code: please just add a `TODO` about this frame skipping algorithm and merge.